### DISCUSSION TITLE
How to extract the referer's header value?

### DISCUSSION THREAD

**Discussion by BenJeau** [tokio-rs/axum#2482] [Q&A] [answered]
### Summary

Hi, I'm using `axum_extra`'s `TypedHeader` to extract the referer, but I'm a bit confused to as how I'm supposed to get it's value. I can log it with it's debug implementation, but not get it's value.

It seems like I need to use `HeaderMap` as an extractor instead of `TypedHeader<Referer>`? I guess I'm just a bit confused as to why I can extract the value for some headers with `TypedHeader`, but not all (or was wondering if it was just a fault from my side).

Thanks :)

### axum version

0.7.2

**Comment by BenJeau** [comment]
After some digging I'm seeing some PRs in the `headers` crate, which I guess can explain this https://github.com/hyperium/headers/pull/95 since I think axum just piggy backs off of that crate?

**Reply by davidpdrsn** [reply, marked as answer]
> since I think axum just piggy backs off of that crate

Exactly. Nothing axum can do about this unfortunately.

**Reply by BenJeau** [reply]
Wanted to make sure I read the source code correctly, thanks for the reply :)
